lib/hypervisor/hv_kvm/__init__.py

```diff
@@ -1987,7 +1987,8 @@ def _ExecuteKVMRuntime(self, instance, kvm_runtime, kvmhelp, incoming=None):
     # during instance startup anyway, and to avoid problems when soft
     # rebooting the instance.
     cpu_pinning = False
-    if up_hvp.get(constants.HV_CPU_MASK, None):
+    if up_hvp.get(constants.HV_CPU_MASK, None) \
+        and up_hvp[constants.HV_CPU_MASK] != constants.CPU_PINNING_ALL:
```

Comment:
To be completely honest, I don't understand why this is necessary and neither the commit message nor the original PR (#1261) help me much. Why is just unsetting `cpu_mask` not enough? @calumcalder, could you please help me out a bit here?

Reply:
Hi @apoikos, as far as I remember we were having issues with dependencies on psutil, which is used by ganeti to set the CPU affinity. This issue was cropping up on configs which had the CPU mask set to CPU_PINNING_ALL, and I narrowed down the issue to CPU pinning code paths still being hit.

I don't remember exactly why unsetting cpu_mask wasn't possible or didn't solve the issue, but I do remember doing some investigation around that (that'll teach me to write better commit messages!)

I can see an argument for having both a PINNING_DISABLED and PINNING_ALL option to be more explicit, and re-enabling the PINNING_ALL option for explicitly setting the CPU affinity to 'any CPU', but the only use case I can see is if for some reason your hypervisor spawns instances with affinities set, and you want to override that, which is not something I've come across.

Reply:
Replying here as well as email reply doesn't put the comments in the right place:

The code already does the dance of using cpu_pinning boolean variable. Changing to unset the HV_CPU_MASK would require changing multiple code paths, versus just adding one more conditional. So to me this is the simplest fix for the problem, which looks good.